Add extraction package

Hchong16 opened the pull request:
Here's the extraction package. Hopefully, my test coverage is sufficient (100% coverage).

Hchong16 committed:
Ability to execute FFMPEG

Hchong16 committed:
Extraction package with test coverage

Hchong16 committed:
Removed unnecessary files

baronj97 commented on extraction/extraction.go:
```diff
@@ -0,0 +1,45 @@
+package extraction
```
Please add a package comment here. https://golang.org/doc/effective_go.html#commentary

Hchong16 replied:
Bad mistake on my part. Will add now.

baronj97 commented on extraction/extraction.go:
```diff
@@ -0,0 +1,45 @@
+	GenerateFrames(n int, video string, dir string) error
+}
+
+// ExtractionService hides the Extractor interface.
```
Since this is an exported struct, let's maybe give some more details. How about something like "ExtractionService provides an interface into generating i-frames from an input video file."

Hchong16 replied:
Got it, will make the change.

baronj97 commented on extraction/extraction.go:
```diff
@@ -0,0 +1,45 @@
+// utilize FFMPEG in order to extract a frame for every n frames in the video and store the
+// ouputs to the dir path.
+func (f FFMPEGExtractor) GenerateFrames(n int, video string, dir string) error {
+	cmd := exec.Command("ffmpeg", "-i", video, "-vf", "select=not(mod(n\\,"+strconv.Itoa(n)+"))",
```
I would replace "ffmpeg" with f.Path.

Hchong16 replied:
Done.

baronj97 commented on extraction/extraction_test.go:
```diff
@@ -0,0 +1,98 @@
+			err := es.IFrames(tc.n, tc.video, tc.dir)
+
+			if err != nil {
+				t.Fatalf("error in iFrames: err = %v", err)
```
Just use `t.Errorf` here. `t.Fatalf` you should only use when setting up a test and something goes wrong, not when evaluating the results

Hchong16 replied:
Gotcha, I wasn't aware of this. Making the change now.

baronj97 reviewed (changes requested):
Overall, looks really good. Just a few minor comments.

Hchong16 committed:
Update package details/comments

Add package commentary, update ExtractionService comment, replace ffmpeg to f.Path, update test cases.

Hchong16 commented:
@baronj97, I resolved all your comments. Let me know if there is anything else that needs to be changed. Thanks!

baronj97 reviewed (approved):
Nice work, LGTM!
